server/src/challenge/challenge.service.ts

```diff
@@ -89,4 +90,48 @@ export class ChallengeService {
       await session.endSession();
     }
   }
+
+  public async update(
+    title: string,
+    request: UpdateChallengeRequestDto
+  ): Promise<Challenge> {
+    const session = await this.connection.startSession();
+    session.startTransaction();
+    try {
+      const challenge = await this.challengeModel
+        .findOne({ title })
+        .populate("solvedUserList")
+        .exec();
+
+      if (!challenge) {
+        throw new Error("Challenge not found");
+      }
+
+      const {
+        title,
+        description,
+        category,
+        fileList,
+        grading,
+        flag,
+      } = request;
+
+      if (title) challenge.title = title;
+      if (description) challenge.description = description;
+      if (category) challenge.category = category;
+      if (fileList) challenge.fileList = fileList;
+      if (grading) challenge.grading = grading;
+      if (flag) challenge.flag = flag;
```

Comment:
```suggestion
      {
        const { title, description, category, fileList, grading, flag } =
          request;

        if (title) challenge.title = title;
        if (description) challenge.description = description;
        if (category) challenge.category = category;
        if (fileList) challenge.fileList = fileList;
        if (grading) challenge.grading = grading;
        if (flag) challenge.flag = flag;
      }
```

함수 매개변수로 온 `title`이랑 request에서 뽑아낸 `title`이 충돌될듯?
`if (title) challenge.title = title` 모양새가 마음에 들어서 그냥 중괄호로 감싸는게 좋다고 생각함